Truststore by default (#11647)
pypa/pip: docs/html/topics/https-certificates.md

By default, pip will perform SSL certificate verification for network
connections it makes over HTTPS. These serve to prevent man-in-the-middle
attacks against package downloads. Pip by default uses a bundled CA certificate

> store from {pypi}`certifi`.

Reviewer: (Don't know what happened to my comment 🙁)

Is this right? Don't we use the system certs (on 3.10+) and the bundled certs now?

Author: My bad, this might have gotten clobbered in the fairly hands-on rebase I had to do to get this branch back in order, let me fix this up! Thank you for catching this.

Author: Check out https://github.com/pypa/pip/pull/11647/commits/e137d6e69ffe5c6dcf4342d31535e047511327a9, happy to restructure things if this isn't clear. I'm hoping for most users things will "just work" and the only thing they need to know about is why they're seeing a different behavior than their coworker.

## Using system certificate stores

```{versionadded} 24.2

```

> If Python 3.10 or later is being used then by default

Reviewer 2: ```suggestion
On Python 3.10 or later, by default
```

Being more concise is probably a good thing. Also using the verb `using` twice in the same sentence is a bit confusing IMO. I would also suggest making a similar change below when discussing Python 3.9 and below. Feel free to reject this suggestion if you disagree.

Author: Updated the wording in https://github.com/pypa/pip/pull/11647/commits/29de6e2a380e39af968b5c98087c6b43f10ac143, thanks for this suggestion!

## Using system certificate stores

```{versionadded} 24.2

```

```{note}
Versions of pip prior to v24.2 did not use system certificates by default.
To use system certificates with pip v22.2 or later, you must opt-in using the `--use-feature=truststore` CLI flag.
```

On Python 3.10 or later, by default
system certificates are used in addition to certifi to verify HTTPS connections.
This functionality is provided through the {pypi}`truststore` package.

If you encounter a TLS/SSL error when using the `truststore` feature you should
open an issue on the [truststore GitHub issue tracker] instead of pip's issue
tracker. The maintainers of truststore will help diagnose and fix the issue.

To opt-out of using system certificates you can pass the `--use-deprecated=legacy-certs`
flag to pip.

```{warning}

> On Python 3.9 or earlier, by default only certifi is used to verify HTTPS connections.

Reviewer 2: ```suggestion
On Python 3.9 or earlier, only certifi is used to verify HTTPS connections as
`truststore` requires Python 3.10 or higher to function.
```

There is no other option to enable truststore on Python 3.9 or earlier, so no need to say "by default." Also explicitly spell out that truststore functionality needs Python 3.10 or higher.

Author: Thank you! I've added your suggested change.
